run/idp-sql/secrets.js

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const { logger } = require('./logging');
+
+// CLOUD_SQL_CREDENTIALS_SECRET is the resource ID of the secret, passed in by environment variable.
+// Format: projects/PROJECT_ID/secrets/SECRET_ID/versions/VERSION
+const {CLOUD_SQL_CREDENTIALS_SECRET} = process.env;
+
+
+// [START run_user_auth_secrets]
+const {SecretManagerServiceClient} = require('@google-cloud/secret-manager');
+let client;
+
+async function getSecrets(secretName) {
+  if (!client) client = new SecretManagerServiceClient();
+  try {
+    const [version] = await client.accessSecretVersion({name: secretName});
+    return version.payload.data;
+  }
+  catch (err) {
+    throw Error(`Error accessing Secret Manager: ${err}`);
+  }
+}
+// [END run_user_auth_secrets]
+
+// Load the Cloud SQL config from Secret Manager
+async function getCredConfig() {
+  if (CLOUD_SQL_CREDENTIALS_SECRET) {
+    const secrets = await getSecrets(CLOUD_SQL_CREDENTIALS_SECRET);
+    // Parse the secret that has been added as a JSON string
+    // to retreive database credentials
+    return JSON.parse(secrets.toString('utf8'));
```

Comment:
I'm not sure what to do here. If I catch it and log and error the app will break anyways since it doesn't have the config. This error will bubble up to the route, so does that mean I should throw an error with a better message?

Reply:
I think a rethrow saying could not parse the secret would be helpful.